projects/Magnetosphere/Magnetosphere.cpp

```diff
@@ -593,182 +594,224 @@ namespace projects {
       return centerPoints;
    }
 
+   bool Magnetosphere::canRefine(const std::array<double,3> xyz, const int refLevel) const {
+      const int bw = (2 + 1*refLevel) * VLASOV_STENCIL_WIDTH; // Seems to be the limit
+
+      return refLevel < P::amrMaxSpatialRefLevel &&
+             xyz[0] > P::xmin + P::dx_ini * bw && 
+             xyz[0] < P::xmax - P::dx_ini * bw && 
+             xyz[1] > P::ymin + P::dy_ini * bw && 
+             xyz[1] < P::ymax - P::dy_ini * bw && 
+             xyz[2] > P::zmin + P::dz_ini * bw &&
+             xyz[2] < P::zmax - P::dz_ini * bw;
+   }
+
```

Comment:
Hmm. It would be good for this function to check against the inner boundary as well. I'm now running a test case where I want to refine the bow shock but not the ionosphere.